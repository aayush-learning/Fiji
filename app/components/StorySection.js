import React, { Component } from 'react';
import { View, Text, Image, ImageBackground } from 'react-native';
import { compose } from 'redux'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { Icon } from 'react-native-elements'
import { fetchStory } from '../redux/story'


class StorySection extends Component {
    constructor(props) {
        super(props)
    }
    render(props) {
        const stories = [];
        console.log("the page number is :", this.props.page)
        var dialog = this.props.story.pages[this.props.page].dialog;
        for (let i = 0; i < this.props.count; i++) {
            for (let j = 0; j <= dialog.length; j++) {

                if (i % 2 === 0) {
                    console.log("this is alice");
                    stories.push(
                        <View style={styles.storyContainer}>
                            <Image style={styles.characterImageStyle}
                                source={{
                                    uri: 'data:image/png;base64,' + this.props.story.characters['Alice'],
                                }} />
                            <View style={styles.textContainer}>
                                <Text style={styles.dialogStyle}>
                                    {this.props.story.pages[this.props.page].dialog[i].text}
                                </Text>
                            </View>
                        </View>
                    );
                    break;
                }
                else {
                    console.log("this is white rabbit");
                    stories.push(
                        <View style={styles.storyContainer2}>
                            <View style={styles.textContainer2}>
                                <Text style={styles.dialogStyle}>
                                    {this.props.story.pages[this.props.page].dialog[i].text}
                                </Text>
                            </View>
                            <Image style={styles.characterImageStyle}
                                source={{
                                    uri: 'data:image/png;base64,' + this.props.story.characters['White Rabbit'],
                                }} />
                        </View>
                    );
                    break;
                }
            }
        }
        //console.log(this.props.story.characters);
        return stories;
    }
}

StorySection.propTypes = {
    story: PropTypes.object,

}
const styles = {
    storyContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        margin: 5

    },
    textContainer: {
        width: 220,
        borderRadius: 20,
        backgroundColor: '#70ef9b',
        padding: 10,
        shadowColor: '#3d3d3d',
        shadowRadius: 2,
        shadowOpacity: 0.5,
        shadowOffset: {
            height: 1,
        },
    },
        textContainer2: {
            width: 220,
            borderRadius: 20,
            backgroundColor: '#f48ded',
            padding: 10,
            shadowColor: '#3d3d3d',
            shadowRadius: 2,
            shadowOpacity: 0.5,
            shadowOffset: {
                height: 1,
            }
    
    },
    characterImageStyle: {
        height: 50,
        width: 50,
        margin: 5,
        borderRadius: 20
    },

    dialogStyle: {
        fontSize: 18,
        color: 'black',
        fontWeight: '600',
    },
    storyContainer2: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        margin: 5,
        alignSelf: 'flex-end',

    },
    nextButtonStyle: {
        flexDirection: 'column',
        justifyContent: 'space-around',
        alignItems: 'center',
        backgroundColor: '#9999ff',
        height: 30,
        shadowColor: 'black',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.9,
        position: 'relative',
        opacity: 1
    }
}

export default connect(state => ({
    story: state.story.story,
    isFetching: state.story.isFetching,
}))(StorySection)
