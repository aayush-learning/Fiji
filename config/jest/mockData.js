import { fetchUsersFailure } from '../../app/redux/users';

const contentJson = require('../seed/content.json')
const usersJson = require('../seed/users.json')
const userAliceJson = require('../seed/user_alice.json')
const storyJson = require('../seed/story.json')

export const titles = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('storytitle:')
})

export const stories = storyJson['docs'].filter(function (doc) {
  return doc._id.startsWith('story:')
})

export const games = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('game:')
})

export const gameThemeData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('gametheme:')
})

export const stickersData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker:caterpillar:')
})

export const stickerPacksData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker-pack:')
})

export const highScoresData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('high-score:')
})

export const story_alice_in_wonderland = storyJson['docs'].filter(function (doc) {
  return doc._id == 'story:alice-in-wonderland'
})[0]

export const users = usersJson['docs']

export const messages = userAliceJson['docs'].filter(function(doc) {
  return doc._id.startsWith('chat:mad-hatter:')
})