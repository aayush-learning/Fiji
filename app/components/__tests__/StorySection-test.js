import 'react-native';
import React from 'react';
import StorySection from '../StorySection';
import { stories } from '../../../config/jest/mockData'

import renderer from 'react-test-renderer';

//npm test -- --updateSnapshot
//this above command is used to update the snapshot in windows

//npm test -- --coverage
//this above command is used to see the total coverage of the tested file in windows


//jest --updateSnapshot
//this above command is used to update the snapshot in IOS

//jest --coverage
//this above command is used to see the total coverage of the tested file in IOS

//npm test -- --coverage User-test
//this above command is used to test indivisual component

it('this is user 0', () => {
  const tree = renderer.create(
  <StorySection
  story={stories}  
  item={stories[1]} 
  index={0} 
  page={0} 
  count={1} 
  bg={stories.pages[0].bg} />
).toJSON();
  expect(tree).toMatchSnapshot();
});


// FAIL  app\components\__tests__\StorySection-test.js
//   × this is user 0 (6ms)

//   ● this is user 0

//     TypeError: Cannot read property '0' of undefined

//       30 |   page={0}
//       31 |   count={1}
//     > 32 |   bg={stories.pages[0].bg} />
//       33 | ).toJSON();
//       34 |   expect(tree).toMatchSnapshot();
//       35 | });

//       at Object.<anonymous> (app/components/__tests__/StorySection-test.js:32:27)

// Test Suites: 1 failed, 1 total
// Tests:       1 failed, 1 total

