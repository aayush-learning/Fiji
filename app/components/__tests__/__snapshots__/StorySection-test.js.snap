// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this is story section 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#09c3f2",
      "flex": 1,
      "height": 70,
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "data:image/png;base64,undefined",
      }
    }
    style={
      Object {
        "height": 100,
        "position": "absolute",
        "width": 200,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 25,
        "fontWeight": "600",
      }
    }
  />
</View>
`;
